internal/testtypes/builders.go

```diff
@@ -363,22 +372,29 @@ func (c *ControlPlaneTemplateBuilder) Build() *unstructured.Unstructured {
 	obj.SetNamespace(c.namespace)
 	obj.SetName(c.name)
 
-	if err := unstructured.SetNestedField(obj.UnstructuredContent(), true, "spec", "template", "spec", "fakeSetting"); err != nil {
-		panic(err)
-	}
+	setSpecFields(obj, c.specFields)
 
 	if c.infrastructureMachineTemplate != nil {
+		// TODO: Update to use new functions in the internal/contract package
```

Comment:
Based on @randomvariable 's suggestion (https://github.com/kubernetes-sigs/cluster-api/pull/5143#discussion_r695651781), I would suggest adding the user-id
```suggestion
		// TODO(killianmuldoon): Update to use new functions in the internal/contract package
```
(same below)